dts/bindings/sensor/ti,tmag5273.yaml

```
@@ -0,0 +1,182 @@
+# Copyright (c) 2023, deveritec GmbH
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+  Texas Instruments TMAG5273 Low-Power Linear 3D Hall-Effect Sensor with an I2C interface.
+  See the specification for the default I2C address.
+  The specification of the sensor can be found at:
+    https://www.ti.com/lit/ds/symlink/tmag5273.pdf
+  The <zephyr/dt-bindings/sensor/tmag5273.h> file should be included in the
+  DeviceTree and it provides macros that can be used for configuration.
+
+compatible: "ti,tmag5273"
+
+include: [sensor-device.yaml, i2c-device.yaml]
+
+properties:
+  operation-mode:
+    type: int
+    default: 0
+    enum:
+      - 0
+      - 1
+    description: |
+      Sensor mode used if set to "active".
+      Defaults to continuous sampling.
+
+        0 # continuous
+        1 # standby
+
+  axis:
+    type: int
+    default: 0x7
+    enum:
+      - 0x0
+      - 0x1
+      - 0x2
+      - 0x4
+      - 0x8
+      - 0x3
+      - 0x5
+      - 0x6
+      - 0x7
+      - 0x8
```

there are weird duplicates in there, and 0x8 should probably not be there either, right?'

BTW, there seems to be hardly any other binding that uses hex notation for enums... not sure if that's a problem?

```suggestion
      - 0x0
      - 0x1
      - 0x2
      - 0x3
      - 0x4
      - 0x5
      - 0x6
      - 0x7
```

---

regarding the hex notation: just used them to have this 1:1 mapping to the datasheet. Not necessary, though.

I am also not sure about the "string" type enums: is this the preferred way, or, as the driver currently does, introducing defines in `include/zephyr/dt-bindings`?

---

> Dang, good find. Something got awry as I was rebasing stuff.
> 
> Well, the sensor is a little bit special, since it states that XYX/YXY/YZY/XZX can be activated (0x8 ... 0xB), but only 3 axis (x,y,z) are available. The datasheet (and tests) aslo didn't reveal anything special here.

It might make sense to document that the "pseudo-simultaneous" sampling options (0x8..0xb) are not supported by the driver? I think these weird "patterns" are what's described at slide 10 here: https://www.tij.co.jp/jp/lit/ml/slyp730/slyp730.pdf#page=10
I.e. XYX would help give you an average reading for the X axis. It is also described in section 8.1.3.3 of the sensor's datasheet (https://www.ti.com/lit/ds/symlink/tmag5273.pdf#page=38)

And maybe you'll want to look into supporting them at some point, it looks like a fun feature :P

---

Actually wasn't aware of that :)